api/src/cctp_calls.rs

```diff
@@ -699,6 +1053,19 @@ pub(crate) fn into_i8(v: Vec<u8>) -> Vec<i8> {
     unsafe { Vec::from_raw_parts(p as *mut i8, len, cap) }
 }
 
+
+struct VerifyProofInputsKeyRotation {
+    cert_data_hash: FieldElement,
+    prev_cert_data_hash: FieldElement,
+    genesis_constant: FieldElement,
+}
+
+impl UserInputs for VerifyProofInputsKeyRotation {
+    fn get_circuit_inputs(&self) -> Result<Vec<FieldElement>, ProvingSystemError> {
+        Ok(vec![self.prev_cert_data_hash, self.cert_data_hash, self.genesis_constant])
+    }
+}
+
```

Comment:
Done

Reply:
Done

Reply:
I see you changed the order but I still see the struct you defined here. You are supposed to use the one in the zendoo-cctp-lib as specified in the first comment in this conversation

Reply:
Can you specify the branch name? My guess is that they are not exactly the same but I can merge it into this.

Reply:
It should be already in the dev branch, you might want to run `cargo update ` to update the dependency and see it (or manually edit the Cargo.lock as you wish)

Reply:
The usage is:

NaiveThresholdSigCircuit: pass sc_prev_wcert_hash as None
NaiveThresholdSigCircuitWKeyRotation: pass Some(PHANTOM_CERT_DATA_HASH) if there is no prev_wcert_hash (current cert is the first one) or, if present, of course pass Some(prev_wcert_hash)

This was done to avoid introducing complex versioning system in mc-cryptolib as we are short on time

Reply:
Done